pkg/apis/servicecatalog/validation/serviceplan.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"regexp"
+
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+// validateServicePlanName is the validation function for ServicePlan names.
+//var validateServicePlanName = apivalidation.NameIsDNSSubdomain
+
+const servicePlanNameFmt string = `[-a-z0-9]+`
+const servicePlanNameMaxLength int = 63
```

Comment:
Where did you get this limit from?

Reply:
This comes from upstream when the serviceplan name was used as a label key, it was failing. 
https://github.com/kubernetes-incubator/service-catalog/issues/1047

I should get rid of that comment above.